geode-core/src/distributedTest/java/org/apache/geode/internal/cache/control/RebalanceOperationDistributedTest.java

```diff
@@ -2834,4 +2778,180 @@ public void close() {
       closed = true;
     }
   }
+
+  private String regionName = "region";
+  private int numOfEntry = 2500;
+  private int totalNumberOfBuckets = 31;
+  private final ArrayList<Integer> list = new ArrayList<>();
```

Comment:
Can this not be a `ConcurrentLinkedQueue` or possibly use `Collections.synchronizedList()`?